terraform/monitoring/04_slos.tf

```diff
@@ -166,4 +166,79 @@ resource "google_monitoring_slo" "istio_service_latency_slo" {
       }
     }
   }
-}
\ No newline at end of file
+}
+
+# Rating service availability SLO:
+# 99% of all non-4XX requests within the past 30 day windowed period
+# return with 200 OK status
+resource "google_monitoring_slo" "rating_service_availability_slo" {
+  # Uses ratingservice service that is automatically detected and created when the service is deployed to App Engine
+  # Identify of the service is built after the following template: gae:${project_id}_servicename
+  service      = "gae:${var.project_id}_ratingservice"
+  slo_id       = "availability-slo"
+  display_name = "Availability SLO with request base SLI (good total ratio)"
+
+  # The goal sets our objective for successful requests over the 30 day rolling window period
+  goal                = 0.99
+  rolling_period_days = 30
+
+  request_based_sli {
+    good_total_ratio {
+
+      # The "good" service is the number of 200 OK responses
+      good_service_filter = join(" AND ", [
+        "metric.type=\"appengine.googleapis.com/http/server/response_count\"",
+        "resource.type=\"gae_app\"",
+        "resource.label.\"version_id\"=\"prod\"",
+        "resource.label.\"module_id\"=\"ratingservice\"",
+        "metric.label.\"response_code\"=\"200\""
+      ])
+
+      # The total is the number of non-4XX and 429 responses
```

Comment:
Please update the comment to mention what is 429 HTTP code - this is mostly because a variety of folks might use this terraform script as a reference documentation on how to implement it for their projects, so it should be as clear as possible

Reply:
Will do